internal/provisioner/kops_provisioner_cluster_installation.go

```diff
@@ -580,15 +614,37 @@ func (provisioner *KopsProvisioner) prepareCILicenseSecret(installation *model.I
 	return licenseSecretName, nil
 }
 
-func (provisioner *KopsProvisioner) deleteLicenseSecret(clusterInstallation *model.ClusterInstallation, k8sClient *k8s.KubeClient, logger log.FieldLogger) error {
-	secretName := fmt.Sprintf("%s-license", makeClusterInstallationName(clusterInstallation))
-	err := k8sClient.Clientset.CoreV1().Secrets(clusterInstallation.Namespace).Delete(context.Background(), secretName, metav1.DeleteOptions{})
-	if k8sErrors.IsNotFound(err) {
-		logger.Warnf("Secret %s/%s not found, assuming already deleted", clusterInstallation.Namespace, secretName)
-		return nil
+func generateCILicenseName(installation *model.Installation, clusterInstallation *model.ClusterInstallation) string {
+	sum := fmt.Sprintf("%x", sha256.Sum256([]byte(installation.License)))
+	if len(sum) > 7 {
```

Comment:
nit: I think it is actually impossible for sha256 to have length different than 32 bytes.

Reply:
Yeah, that was a large amount of extra caution on my part, but that's probably completely unnecessary so let me remove it.